yt/data_objects/static_output.py

```diff
@@ -341,12 +338,19 @@ def force_periodicity(self, val=True):
         # workaround in yt user codes.
         if not isinstance(val, bool):
             raise TypeError("force_periodicity expected a boolean.")
-        self._force_periodicity = val
+        if val is True:
+            self.set_periodicity(True)
+        else:
+            self.set_periodicity(None)
```

Comment:
`force_periodicity` is meant to lock the observable state of `self.periodicity`, reseting defaults with `None` isn't among the things it can do. For full backward compatibility I think it should simply be this
```suggestion
        self.set_periodicity(val)
```

Reply:
Yes, that makes sense! I should have realized that.

Reply:
In insight I understand it is not obvious, `force_periodicity(False)` is ambiguous from the outside.
I think your `set_periodicity` is much clearer in what it does. Btw I encourage you to grep for `force_periodicity` throughout the code base.
In most places I think you could just replace it with `set_periocity`, especially in documentation. In one case (`doc/source/cookbook/hse_field.py`) it looks like you could just get rid of it.